### PR TITLE
Tests & interface changes

### PR DESCRIPTION
**Pull request by gaorlov**
- Adding tests. Up to ~97% coverage. planning on going to 100% with a massive refactor. 
## BREAKING changes
- changing `property` to only take a single property and optional default
- removing `all_without_pagination`
- removing support for declaring the client class as 

``` ruby
class Resource < JsonApiResource::Resource
  def self.client_klass = Client
end
```
## Interface changes
- `properties` is the way to add many properties (as a hash)

``` ruby
  properties id: nil,
             name: ""
```
- `property` is the way to add a single property. `nil` is the default

``` ruby
  property :id
  property :name, ""
```
- `wraps` is the way to declare the `JsonApiClient` class the resource wraps

``` ruby
  wraps Client
```

### CODE REVIEW

File: lib/json_api_resource/conversions.rb [hunk 1]
```diff
@@ -55,19 +43,12 @@ def Boolean(*args)
       end
     end
 
-    def Symbolize(*args)
-      case args.first
-        when String then args.first.underscore.parameterize('_').to_sym
-        else args.first
-      end
-    end
```
Comment (gaorlov) [inline lib/json_api_resource/conversions.rb:63]: What is this? Is this a safe `to_sym`? Do we need this?

File: lib/json_api_resource/schemable.rb [hunk 2]
```diff
@@ -9,11 +9,16 @@ module Schemable
     end
 
     module ClassMethods
-      def property(opts = {})
-        opts.each do |attr_name,default|
-          self.schema[attr_name.to_sym] = default || nil
```
Comment (gaorlov) [inline lib/json_api_resource/schemable.rb:14]: interface change here.

File: lib/json_api_resource/resource.rb [hunk 3]
```diff
@@ -1,3 +1,12 @@
+require 'active_support'
+require 'active_support/concern'
+require 'active_support/core_ext/module'
+require 'active_support/core_ext/class/attribute'
+require 'active_model'
+require 'active_model/model'
+require 'active_model/validations'
+require 'active_model/callbacks'
```
Comment (bluetwin) [inline lib/json_api_resource/resource.rb:9]: Why do we need this? I'm not sure if this is already tied to rails, but do we need to tie this directly to rails?

File: test/test_helper.rb [hunk 4]
```diff
@@ -1,9 +1,64 @@
...
+  class << self
+    def client_klass
+      UserClient
+    end
+
+    def schema
+      { id: 0,
+        name: ""
+      }
+    end
```
Comment (bluetwin) [inline test/test_helper.rb:40]: what is the difference between schema and the properties?

File: test/test_helper.rb [hunk 5]
```diff
@@ -1,9 +1,64 @@
...
-class User < JsonApiClient::Resource
+class UserClient < JsonApiClient::Resource
   self.site = "http://localhost:3000/api/1"
 end
+
+class UserResource < JsonApiResource::Resource
+  class << self
+    def client_klass
+      UserClient
+    end
```
Comment (bluetwin) [inline test/test_helper.rb:34]: should we change this to:

``` ruby
class UserResource < JsonApiResource::Resource
  client_klass UserClient
end
```

File: lib/json_api_resource/resource.rb [hunk 6]
```diff
@@ -1,3 +1,12 @@
...
+
 module JsonApiResource
   class Resource
     include ActiveModel::Model
```
Comment (bluetwin) [inline lib/json_api_resource/resource.rb:13]: i guess this `ties it to rails. should we?

Comment (gaorlov) [inline lib/json_api_resource/resource.rb:13]: I think that's probably fine. I don't know of a usecase for this kind of object outside of a web framework.

File: test/test_helper.rb (hunk 5 above)
Comment (gaorlov) [inline test/test_helper.rb:34]: Yeah. Maybe 

``` ruby
class UserResource < JsonApiResource::Resource
  wraps_client UserClient
end
```

or simply

``` ruby
class UserResource < JsonApiResource::Resource
  client UserClient
end
```

File: test/test_helper.rb (hunk 4 above)
Comment (gaorlov) [inline test/test_helper.rb:40]: It's the existing paradigm. I would like to move away from it. Maybe we might as well now that I broke `property`

File: test/test_helper.rb (hunk 5 above)
Comment (gaorlov) [inline test/test_helper.rb:34]: I ended up going with

``` ruby
class UserResource < JsonApiResource::Resource
  wraps UserClient
end
```

What do you think of that?

File: lib/json_api_resource/resource.rb (hunk 6 above)
Comment (bluetwin) [inline lib/json_api_resource/resource.rb:13]: I wonder if anyone may need this in a rack app or sinatra app?

Comment (gaorlov) [inline lib/json_api_resource/resource.rb:13]: Then they can open an issue or MR. Concerns and am::m are things we rely heavily on

**Pull request merged**